src/CylindricalMagnetChain_geo.cpp

```diff
@@ -45,9 +46,14 @@ static Ref_t create_magnet(Detector& description, xml_h e, SensitiveDetector /*
     Tube yoke_tube(rin, rout, 0.5 * length);
     Volume v_yoke("v_yoke_" + name, yoke_tube, m_Iron);
 
-    v_yoke.setVisAttributes(description.visAttributes(vis_name));
+    v_yoke.setVisAttributes(x_det.visStr());
 
-    assembly.placeVolume(v_yoke, Transform3D(RotationY(theta), Position(x, y, z)));
+    auto yoke_pv = assembly.placeVolume(v_yoke, Transform3D(RotationY(theta), Position(x, y, z)));
+
+    yoke_pv.addPhysVolID("element", id);
+    DetElement yoke_de(sdet, name, id);
+    yoke_de.setPlacement(yoke_pv);
+    yoke_de.setAttributes(description, v_yoke, x_det.regionStr(), x_det.limitsStr(), vis_name);
```

Comment:
I'd still like to avoid this, if we can. We only have 255 detector IDs in our 8-bit field for segmentations, and less than half are left. If we start giving IDs to all magnets, we're going to run out quickly.

Note that you can add regions and limits to volumes too:
```
DDCore/include/DD4hep/Volumes.h:
    const Volume& setAttributes(const Detector& description, const std::string& region, const std::string& limits,
                                const std::string& vis) const;
```

Can you verify that visualization is not affected when avoiding the ID for this?

Reply:
I have removed the DetElement and recovered the visualization using setAttributes. The physical volume ID is still required to give the volume a unique identifier in the tree but shouldn't count towards the detector IDs